File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.py
def solution(data, ext, val_ext, sort_by):
    answer = [[]]
    data_type = {"code" : 0, "date" : 1, "maximum" : 2, "remain" : 3}
    filtered_data = [d for d in data if d[data_type[ext]] < val_ext]

    sorted_filtered_data = sorted(filtered_data, key=lambda x:x[data_type[sort_by]])
    return sorted_filtered_data

File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.java
import java.util.*;

class Solution {
    public int[][] solution(int[][] data, String ext, int val_ext, String sort_by) {
        int filterIndex = indexTranslator(ext);
        
        
        return Arrays.stream(data).filter(x -> x[filterIndex] < val_ext).sorted((o1, o2) -> o1[indexTranslator(sort_by)] - o2[indexTranslator(sort_by)]).toArray(int[][]::new);
    }
    
    private int indexTranslator(String ext) {
        switch (ext) {
            case "code" : return 0;
            case "date" : return 1;
            case "maximum" : return 2;
            default : return 3;
        }
    }
}